### PR TITLE
feat: add time subtitle to event list

### PR DESCRIPTION
**Pull request by ardasnturk**
- added `withTime` option to `listItemParser` to show the event time in the event list
- added `withTime` and `dates` control to `listItemParser` to show the timing of the event in the list
- added `second` prop to `textHelper` to show the time in the list in the desired format and updated returns

SVAK-84

## Screenshots:

|with time|without time|
|--|--|
Simulator Screenshot - iPhone 15 Pro Max - 2024-05-06 at 12 55 36|Simulator Screenshot - iPhone 15 Pro Max - 2024-05-06 at 12 55 47

### CODE REVIEW

File: src/helpers/parser/listItemParser.js [hunk 1]
```diff
@@ -43,11 +43,12 @@ const filterGenericItems = (item) => {
   return true;
 };
 
-const parseEventRecords = (data, skipLastDivider, withDate) => {
+const parseEventRecords = (data, skipLastDivider, withDate, withTime) => {
   return data?.map((eventRecord, index) => ({
     id: eventRecord.id,
     subtitle: subtitle(
       withDate ? eventDate(eventRecord.listDate) : undefined,
+      withTime && !!eventRecord?.dates?.[0].timeFrom ? eventRecord?.dates?.[0].timeFrom : undefined,
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/helpers/parser/listItemParser.js:51]: Yes, that's right! I'll fix it. 👍

File: src/helpers/textHelper.js [hunk 2]
```diff
@@ -2,16 +2,21 @@
...
  * @param {string | undefined} first text before the pipe
+ * @param {string | undefined} second text before the pipe (Uhr)
  * @param {string | undefined} last text behind the pipe
  *
- * @return {string} a formatted string `first | last` or `first` or `last`
+ * @return {string} a formatted string `first | last` or `second Uhr, last` or `second Uhr` or `first` or `last`
  */
-export const subtitle = (first, last) => {
-  if (!first && !last) return '';
+export const subtitle = (first, second, last) => {
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/helpers/textHelper.js:10]: Yes, you are right. It's not safe. As you said, we use it in other places and we don't want to change the structure there.

[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/helpers/textHelper.js]: https://github.com/smart-village-solutions/smart-village-app-app/commit/5ef913e33b6cef0d1402510b4db56328b51e3224

File: src/helpers/parser/listItemParser.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/helpers/parser/listItemParser.js]: https://github.com/smart-village-solutions/smart-village-app-app/commit/5ef913e33b6cef0d1402510b4db56328b51e3224

[the remark this answers is not in the document]
Comment (donni106) [reply src/helpers/parser/listItemParser.js]: why did you choose a different code than i suggested? copy & paste would have been fine.
now in your change it is not "value or undefined" but also can be a boolean false.

[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/helpers/parser/listItemParser.js]: I'm sorry. I didn't read your comment correctly. I'm updating the code.

[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/helpers/parser/listItemParser.js]: https://github.com/smart-village-solutions/smart-village-app-app/commit/c424c4b5d68ad936bce5aad06a144f10cc172429

**Review by donni106 — APPROVED** [review]

**Pull request merged**